main.tf

```diff
@@ -1 +1,346 @@
+  load_balancer_arn = "${ aws_lb.this.arn }"
+  port              = "443"
+  protocol          = "HTTPS"
+  ssl_policy        = "ELBSecurityPolicy-FS-2018-06"
+  certificate_arn   = "${var.lb_certificate_arn}"
+
+  default_action {
+    target_group_arn = "${ aws_lb_target_group.this.arn }"
+    type             = "forward"
+  }
+}
+
+resource "aws_lb_target_group" "this" {
+  name     = "${var.name}-${var.environment}"
+  port     = "8200"
+  protocol = "HTTP"
+  vpc_id   = "${ local.vpc_id }"
+
+  deregistration_delay = "10"
+
+  # /sys/health will return 200 only if the vault instance
+  # is the leader. Meaning there will only ever be one healthy
+  # instance, but a failure will cause a new instance to
+  # be healthy automatically. This healthceck path prevents
+  # unnecessary redirect loops by not sending traffic to
+  # followers, which always just route traffic to the master
+  health_check {
+    path                = "/v1/sys/health?standbyok=true"
+    port                = "8200"
+    interval            = "5"
+    timeout             = "3"
+    healthy_threshold   = "2"
+    unhealthy_threshold = "2"
+  }
+
+  tags = "${ merge(
+    map("Name","${ var.name }-tg"),
+    local.tags ) }"
+}
+
+#Security Groups
+resource "aws_security_group" "lb" {
+  name        = "${var.name}-${var.environment}"
+  description = "Rules required for operation of ${var.name}"
+  vpc_id      = "${local.vpc_id}"
+
+  ingress {
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = "${ merge(map("Name", "${var.name}-lb-${var.environment}"), local.tags ) }"
+}
+
+resource "aws_security_group" "ec2" {
+  name        = "${var.name}-ec2-sg-${var.environment}"
+  description = "Rules required for operation of ${var.name}"
+  vpc_id      = "${local.vpc_id}"
+
+  ingress {
+    from_port       = 8200
+    to_port         = 8200
+    protocol        = "tcp"
+    security_groups = ["${aws_security_group.lb.id}"]
+  }
+
+  ingress {
+    from_port = 8201
+    to_port   = 8201
+    protocol  = "tcp"
+    self      = true
+  }
+
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["${local.allow_inbound}"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = "${ merge(map("Name", "${var.name}-ec2-sg-${var.environment}"), local.tags ) }"
+}
+
+# Manage autoscaling group
+locals {
+  appscript_url    = "s3://${local.bucket}/${random_string.this.result}/scripts/appscript.sh"
+  appscript_params = "${local.bucket}/${random_string.this.result} ${var.vault_version} ${var.dynamodb_table} ${data.aws_kms_key.this.key_id} ${local.ssm_root_path}"
```

Comment:
This might be easier to comprehend if constructed as a list here, and then joined when passed to the cfn resource